### DISCUSSION TITLE
Having a machanism to rewrite res:// paths when mounting pck

### DISCUSSION THREAD

**Discussion by TML233** [godotengine/godot-proposals#11021] [Engine Core]
To implement mounting pcks under a specific resource path, we need to find a way of rewriting `res://` path references when mounting pck. This makes the `res://` path references in the exported pck still being valid when they are mounted under a specific resource path in a running game.
Say you have a base game, and it mounts a `mod_foo.pck` file under its `res://mods/foo/` folder. The file `res://image.png` of the modpack are now placed under `res://mods/foo/image.png` in the base game. So when a scene in the modpack accesses `res://image.png`, the access path needs to be rewritten to `res://mods/foo/image.png` to ensure the resource dependencies of the modpack still integrated. And the scripts in the modpack can also hard-code `res://` paths in the code.

**Comment by Sasino97** [comment]
Yes, we need this.
